cogs/groupsCog.py

```diff
@@ -4,288 +4,211 @@
-            role = discord.utils.get(ctx.guild.roles, name=group_name)
-            if role:
-                await role.delete()
-                logging.debug(f"Deleted role {role}.")
-
-            # remove channels
-            for channel_id in channels:
-                tmp_channel = ctx.guild.get_channel(channel_id)
-                if tmp_channel:
-                    await tmp_channel.delete()
-                    organize_group.remove_channel(sanitized_guild, group_name, tmp_channel.id)
-                    logging.debug(f"Deleted channel {tmp_channel.name}.")
-
-            # remove category
-            tmp_category = ctx.guild.get_channel(category)
-            if tmp_category:
-                await tmp_category.delete()
-                logging.debug(f"Deleted category {tmp_category}.")
-        for player_id in players:
-            member = ctx.guild.get_member(player_id)
-            await member.send(msg["group_deleted_info"].format(author=ctx.author, name=group_name))
-        await ctx.reply(result_str)
-
-    @commands.command(help=msg["gdestroy_help"],
-                      aliases=['gentfernen'])
+        group = Group(
+            name=name,  # slug generated from name
+            date=time,
+            max_players=maximum_players,
+            description=description,
+            ctx=ctx,  # sets guild id
+            owner=ctx.author.id,
+            bot=self.bot,
+        )
+        if group.exists:
+            await ctx.reply(msg["gcreate_group_exists"].format(name=name), ephemeral=True)
+            return
+        await group.setup_guild()  # create role, category and channels
+        group.save()  # writes to yaml
+        content = msg["gcreate_success"].format(group=group)
+        buttons = group.info_view(ctx.author)
+        group.message = await ctx.reply(content, view=buttons)
+
+    @gcreate.error
+    async def gcreate_error(self, ctx, error):
+        if (isinstance(error, commands.errors.MissingRequiredArgument) 
+        or isinstance(error, commands.errors.BadArgument)):
+            view = NewGroupView(ctx.author)
+            view.bot = self.bot
+            await ctx.send(msg["gcreate_bad_args"].format(
+                pre=ctx.prefix, cmd=ctx.command.name, sig=ctx.command.signature
+            ), view=view)
+            # raise commands.CommandInvokeError(error)  # prevent propagation
+
+    @command(help=msg["glist_help"], aliases=['gliste'])
+    async def glist(self, ctx, full: bool = param(
+            default=False, description=msg["glist_desc"])
+        ):
+        await ctx.reply(msg["glist_header"])
+        for group in Group.groups_from_guild(ctx.guild.id):
+            group.bot = self.bot  # pass bot so group can fetch data form discord api.
+            group.message = await ctx.reply(
+                group.info_message, view=group.info_view(ctx.author))
+
+    @command(help=msg["gedit_help"], aliases=['gbearbeiten'])
+    async def gedit(self, ctx,
```

Comment:
`TypeError: argument of type 'NoneType' is not iterable`
when there is no group